Bindings/Java/Matlab/tests/testOsimC3D.m

```diff
@@ -34,76 +34,92 @@
+        continue
+    elseif diff_array(2) ~= 0 || diff_array(3) ~= 0
+        error(['Error in rotateData(). Marker rotations about X are incorrect'])
     end
 end
-for u = 1 : nFlabels
-    % Get the sum of the differences between original and rotated data
-    p = nansum(forceRef180X.(flabels{u}) + forceRef180Y.(flabels{u}));
-    % Sum should be at or near zero. Round to 5 decimal places 
-    if round(p(1),5) ~= 0 | round(p(3),5) ~= 0
-        error(['Error in rotateData(). Rotations about Y are incorrect'])
-    end
- end
 
-% Rotate Data 180 degrees around Z and check that X and Y data are
-% mirrored
-c3d.rotateData('z',-90);c3d.rotateData('z',-90);
-[markerRef180Z, forceRef180Z] = c3d.getAsStructs;
-% For each marker, Y and Z data should be mirrored when rotating 180
-% about x. 
-for u = 1 : nMlabels
-    % Get the sum of the differences between original and rotated data
-    p = nansum(markerRef180Z.(mlabels{u}) + markerRef180Y.(mlabels{u}));
-    % Sum should be at or near zero. Round to 5 decimal places 
-    if round(p(1),5) ~= 0 | round(p(2),5) ~= 0
-        error(['Error in rotateData(). Rotations about Z are incorrect'])
+% Test Force Rotations
+randomRows = randsample(forces.getNumRows(),6)-1;
+randomCols = randsample(forces.getNumColumns(),6)-1;
+for i = 1 : 6
+    % Get the values in the original and rotated marker tables
+    val = osimVec3ToArray(forces.getRowAtIndex(randomRows(i)).get(randomCols(i)));
+    val_rotated = osimVec3ToArray(forces_rotated.getRowAtIndex(randomRows(i)).get(randomCols(i)));
+    % Compute the difference in the marker locations. The Y and Z values
+    % should be mirroed and result in 0 when added. Round to 8 decimal
+    % places. 
+    diff_array = round(val + val_rotated,8);
+    % Compare the difference
+    if isnan(diff_array)
+        % If the value is a nan then move the next value
+        continue
+    elseif diff_array(2) ~= 0 || diff_array(3) ~= 0
+        error(['Error in rotateData(). Force rotations about X are incorrect'])
     end
 end
-for u = 1 : nFlabels
-    % Get the sum of the differences between original and rotated data
-    p = nansum(forceRef180Z.(flabels{u}) + forceRef180Y.(flabels{u}));
-    % Sum should be at or near zero. Round to 5 decimal places 
-    if round(p(1),5) ~= 0 | round(p(2),5) ~= 0
-        error(['Error in rotateData(). Rotations about Z are incorrect'])
+
+%% Test mm to M conversion for forces
+c3d.convertMillimeters2Meters();
+forces_m = c3d.getTable_forces();
+
+% Forces should remain unchanged, point and moments should be in meters
+% (divided by 1000)
+randomRows = randsample(forces.getNumRows(),101)-1;
+for i =  0 : forces_m.getNumColumns() - 1
+    
+    col =  forces_rotated.getDependentColumnAtIndex(i);
+    col_m = forces_m.getDependentColumnAtIndex(i);
+    label = forces_m.getColumnLabel(i);
+    
+    for u = 1 : length(randomRows)
+        if contains(char(forces_m.getColumnLabels().get(i)),'f')
+            % Get the values
```

Comment:
Should this at least be start with 'f' rather than contains 'f'? Not convinced that making up a convention while reading data is the right thing to do, but seems unrelated.

Reply:
I preferred _contains()_  in this case because it will return a 1 or 0. If I use something like _strfind()_ it will return either 1 or [], which means I have to do something like ~isempty(strfind()).  Since we (_C3DFileAdapter()_) are the ones that define the column names, we are guaranteed of not having a case when there may be a rogue _f_ in the column name (like if someone named their column COP_f for some reason).

Reply:
ok, if there's nothing like startsWith() that returns logical.

Reply:
Oh, SNAP! hahaha, I hadn't seen that! Thanks! Perfect I will change to startsWith()